lang/go/code-common-functions.talon-list

```diff
@@ -0,0 +1,20 @@
+list: user.code_common_function
```

Comment:
This it the list of predeclared functions, but many of them aren't very common. (Also: `make`?)

These are the top ten identifiers across a pretty large Go corpus:

```
  4.62%   4.62% 10517712 err
  3.14%   7.75% 7149343 nil
  2.64%  10.40% 6019367 string
  2.57%  12.97% 5859233 t
  1.13%  14.09% 2564886 error
  0.99%  15.09% 2267076 c
  0.91%  16.00% 2081426 name
  0.87%  16.87% 1982235 s
  0.83%  17.70% 1881744 _
  0.81%  18.50% 1835017 ctx
```

len is #16 in the list. append is #35. close is #96. println is #197 (and note that that includes uses of fmt.Println!). clear is #1725. imag doesn't even make [the top 5000](https://gist.github.com/josharian/593ff6dc72e249d838c20484c5b61550).

I'd rather not worry about the built-ins and choose based on use.

My 2c: I'd keep len, append, close, cap, clear, delete, max, min, new, panic, and recover. And ditch the rest.

And consider maybe adding important std calls, such as `errors.New`, `fmt.Errorf`, `fmt.Printf`, and the like. If it'd be helpful, I can pull usage stats on those.

FWIW, here's what I have for package fmt (some of which are pretty metal):

```
# fmt
fmt dot print: fmt.Print
thumped up print: fmt.Print

fmt dot print line: fmt.Println
thumped up print line: fmt.Println

fmt dot print ef: fmt.Printf
thumped up print ef: fmt.Printf
fmt dot print dev: fmt.Printf
thumped up print dev: fmt.Printf
fmt dot print death: fmt.Printf
thumped up print death: fmt.Printf

# # this is how conformer D spells/hears "error eff"
fmt dot erf: fmt.Errorf
thumped up erf: fmt.Errorf

fmt dot sprintf: fmt.Sprintf
thumped up sprintf: fmt.Sprintf
fmt dot sprint dev: fmt.Sprintf
thumped up sprint dev: fmt.Sprintf
fmt dot sprint death: fmt.Sprintf
thumped up sprint death: fmt.Sprintf
```

Also, there are common identifiers that are hard to pronounce but aren't function calls, like `req`, `resp`, `cmd`. How do those fit in here?

Reply:
apologies if i'm misunderstanding the context here

'make' is necessary to initialize maps, slices, chans. its pretty essential

what is the Go corpus you looked at?

Reply:
Yes, my parenthetical was saying we need `make`. :)

The corpus: https://github.com/mvdan/corpus

Reply:
oh man, i worried about that, sorry!
kinda interesting make doesnt show up that often then?

Reply:
Yeah. In particular, I'm surprised that `new` shows up more than `make`.

Of course, that's a list of _all_ identifiers, not just functions.